### PR TITLE
Debug - 'Your Files' tab does not display user uploaded files

### PR DESCRIPTION
**Pull request by Sathya-S3**
The error was caused after the introduction of the Boolean search feature, where the search specifications dictionary is expected to include a 'boolean' flag to indicate choice of an AND/NOT operation for the search. Absence of this flag in the search query generated from the FileListerController caused inaction in search.py, preventing the generation of search query using user's ORCID details.

By adding the 'boolean' flag to the search specifications dictionary, this issue has been solved.

Additionally, I have performed a minor code change in search.py where I have replaced 'if bool_inspect == false' and 'bool_inspect == true' with 'if not bool_inspect' and 'if bool_inspect' respectively, where equality comparisons to Boolean values have been replaced with a more appropriate syntax.

### CODE REVIEW

**Comment by Sathya-S3** [comment]
@jkshenton CI tests failing seems strange. The failed tests do not seem to be related to the code changes in this pull request. This seems to be related to the gridfs version used by GitHub for its checks. The pytest tests passed successfully on the server.

**Comment by jkshenton** [comment]
Thanks Sathya! I've suggested a few tweaks. For the CI, is it worth tinkering with the `setup.py` or the Action to get the right version of gridfs/mongomock or whatever is not working?

**Comment by Sathya-S3** [comment]
@jkshenton In the latest commit 0841df6, I have renamed 'boolean' to 'negate_query' for clarity in both the .py and .js scripts. 

The manual Boolean choice choice reporting for Your Files has been removed, so the AND condition for query construction works both when the negate_query if False or None. The if condition does not succeed by default when negate_query is None, so the if condition is changed to 'if negate_query is False or negate_query is None:'. The elif is replaced by else. All searches combinations checked and found to be working. Your Files displays uploads as it should.

**Comment by Sathya-S3** [comment]
@jkshenton The CI error seems to originate from enable_gridfs_integration() which had been working fine in previous pull requests, including when when we upgraded the dependency versions. This function is from mongomock.gridfs of the python site packages, which patches pymongo to mongomock database as:
`mock.patch('gridfs.Database', (PyMongoDatabase, MongoMockDatabase)).start()`
which seems to the origin of error, as the attribute 'Database' is reported missing. This may be a problem arising from more recent versions of mongomock that pytest uses. 

What I tried: I tried changing mongomock version to ~=4.0 in setup.cfg and requirements.txt as it worked before. But, pytest still seems to install mongomock 4.2 for the tests, which may or may not be the source of the issue.

An alternative that I have verified to work locally (not yet on GitHub) is to patch the enable_gridfs_integration() function with custom mock classes to simulate the GridFS system behaviour in our unit tests. This is a way to still run tests without a complete rewrite of the test code, as we still need to use mongomock functions within gridfs. I'll attempt to push this change to see if this fixes things on GitHub. We can chat again before merging anyway.

**Comment by jkshenton** [comment]
Thanks Sathya!  

For the CI tests, I would be quite hesitant to patch mongomock in that way when there should be a version that works. Your test for changing the version number didn't work because of the `~` in the specification. That means that it would install the minor latest version up to but not including the next major release - i.e. it would install v4.2 even if you set `~=4.0`. What version are you running on your machine? You could try pinning to `mongomock>=4.1,<4.2` if 4.1 works but 4.2 doesn't, for example. 

Otherwise figuring out what changed in the newer version of mongomock that causes the tests to fail would be good - it might suggest a better way of doing our tests anyway... ?

File: ccpncdb/search.py [hunk 1]
```diff
@@ -309,12 +309,12 @@ def build_search(search_spec):
         query_buffer = search_func(**args)
         query_add=[]
         # Make choice to pass query as such or negate it based on user's Boolean filtering choice in 'boolean' key
-        if bool_inspect == False: #AND - pass query as is
+        if negate_query is False or negate_query is None: #AND - pass query as is
```
[the remark this answers is not in the document]
Comment (Sathya-S3) [reply ccpncdb/search.py]: This is done now! Verified to work on localhost as well.

**Comment by Sathya-S3** [comment]
> Thanks Sathya!
> 
> For the CI tests, I would be quite hesitant to patch mongomock in that way when there should be a version that works. Your test for changing the version number didn't work because of the `~` in the specification. That means that it would install the minor latest version up to but not including the next major release - i.e. it would install v4.2 even if you set `~=4.0`. What version are you running on your machine? You could try pinning to `mongomock>=4.1,<4.2` if 4.1 works but 4.2 doesn't, for example.
> 
> Otherwise figuring out what changed in the newer version of mongomock that causes the tests to fail would be good - it might suggest a better way of doing our tests anyway... ?

You have a valid point. The error seems to originate from pymongo's gridfs library rather than mongomock itself. So, I have lowered pymongo version to >=3.12.3,<4.0, which is more compatible with mongomock's enable_gridfs_integration() function which in itself is a patch.

I have also reverted the custom patch that I had applied. Everything works as it should.

**Review by jkshenton — APPROVED** [review]
This now looks ready to merge - thanks again Sathya!

**Pull request merged**